bin/scrapeReference42021.js

```diff
@@ -1,122 +1,123 @@
+ * @param {string} href 
+ * @param {string} title 
+ * @param {string | null} text 
+ * @returns {string}
+ */
+renderer.link = function( href, title, text ) {
+	return `<a  href="https://accessibilite.numerique.gouv.fr/methode/glossaire/${href}" target="_blank" title="${title ?? text}">${text}</a>`;
+  }
 
 /**
- *	Scrapes version 4 of the RGAA reference into JSON.
+ *	Transforms data from version 4.1 of the RGAA reference into JSON.
  *
  *	@param {object} options - Options:
  *		- {string} source - Source URL.
  *		- {string} destination - Destination file.
  *		- {boolean} merge - Whether or not to merge the output
- *			file with the existing one, if any.
+ *		- file with the existing one, if any.
  */
-module.exports = (options) => (html) => {
-	const $ = cheerio.load(html, {
-		normalizeWhitespace: true,
-		decodeEntities: false
-	});
 
-	installResolveLinksPlugin($);
-	installRemoveChildrenPlugin($);
-
-	const scrapeTest = (i, el) => {
-		const element = $(el);
-		const title = element
-			.resolveLinks(options.source)
-			.html()
-			.trim();
-		const idMatches = /^<span class="test">Test (\d+\.\d+\.\d+)/i.exec(title);
-
-		if (idMatches === null) {
-			return null;
+module.exports = (options) => (json) => {
+	try {
+		if (!json || typeof json !== 'string') {
+			throw new Error('RGAA Criteria  Json file missing');
 		}
+		const rgaaJson = JSON.parse(json);
+
+		return {
+			name: 'RGAA 4.1 (2021)',
+			version: '4-2021',
+			themes: buildThemes(rgaaJson)
+		};
+	} catch (error) {
+		throw new Error(error.message);
+	}
+};
 
-		const id = idMatches[1];
-
-		let formattedTitle = title.replace(
-			/^<span class="test">Test \d+\.\d+\.\d+<\/span>&#160;:\s/i,
-			''
-		);
-
-		if (element.find('ul').length) {
-			let conditions = element.find('ul')
-				.resolveLinks(options.source)
-				.html()
-				.trim();
-			conditions = `<ul>${conditions}</ul>`;
-			formattedTitle = formattedTitle.replace(
-				/<ul>(.*)<\/ul>/,
-				''
-			);
-			formattedTitle = `${formattedTitle}${conditions}`;
-		} else {
-			formattedTitle = `${formattedTitle}`;
-		}
-
-		return {id, title: formattedTitle};
-	};
-
-	const scrapeCriterion = (i, el) => {
-		const element = $(el);
-		const title = element
-			.find('h4')
-			.removeChildren('button')
-			.resolveLinks(options.source)
-			.html()
-			.trim();
-
-		const idMatches = /^Critère (\d+)\.(\d+)/i.exec(title);
-
-		if (idMatches === null) {
-			return null;
-		}
-
-		const id = `${idMatches[1]}.${idMatches[2]}`;
-
-		const level = element.attr('data-level');
-
-		const formattedTitle = title.replace(
-			/^Critère \d+\.\d+\. /i,
-			''
+/**
+ * @param {object} rgaaJsonCriteria
+ * @returns {Array} Array
+ */
+function buildThemes(rgaaJsonCriteria) {
+	try {
+	let themes = [];
+	rgaaJsonCriteria.topics.map((topic) => {
+		themes = [
+			...themes,
+			{
+				'id': `${topic.number}`,
+				'title': `${topic.number}. ${topic.topic}`,
+				'criteria': parseCriteria(topic.criteria, topic.number)
+			}
+		];
+	});
```

Comment:
Le map est un peu étrange ici, il ressemble plutôt à un forEach, mais plus bas tu l'as bien fait.
```suggestion
	return rgaaJsonCriteria.topics.map((topic) => ({
		'id': `${topic.number}`,
		'title': `${topic.number}. ${topic.topic}`,
		'criteria': parseCriteria(topic.criteria, topic.number)
	}));
```